ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Authorizer.scala

```diff
@@ -170,6 +172,26 @@ final class Authorizer(
       call,
     )
 
+  def authenticatedUserId(): Try[Option[String]] =
+    authenticatedClaimsFromContext()
+      .flatMap(claims =>
+        if (claims.resolvedFromUser)
+          claims.applicationId match {
+            case Some(applicationId) => Success(Some(applicationId))
+            case None =>
+              Failure(
+                LedgerApiErrors.AuthorizationChecks.InternalAuthorizationError
+                  .Reject(
+                    "unexpectedly the user-id is not set in the authenticated claims",
```

Comment:
```suggestion
                    "the user-id is not set in the authenticated claims",
```

Reply:
The `unexpectedly` is there to remind the developer that this is not expected. It is an internal error.